zstacklib/zstacklib/utils/gpu.py

```
@@ -661,6 +662,37 @@ def has_nvidia_gpu():
+def exec_sriov_manage(addr, is_enable=True):
+    if is_enable:
+        return bash_roe("/usr/lib/nvidia/sriov-manage -e %s" % addr)
+    else:
+        return bash_roe("/usr/lib/nvidia/sriov-manage -d %s" % addr)
+
+
+def detach_from_host(addr, vendor):
+    if os.path.exists('/usr/lib/nvidia/sriov-manage'):
+        ret, out, err = exec_sriov_manage(addr, False)
+        if ret != 0:
+            raise Exception("failed to /usr/lib/nvidia/sriov-manage -d %s: %s, %s" % (addr, out, err))
```

Ruff (0.14.8)

Create your own exception

(TRY002)